src/Raven.Pal/src/posix/spawn.c

```diff
@@ -0,0 +1,291 @@
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+
+#include <sys/wait.h>
+#include <pthread.h>
+#include <sys/time.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <sys/mman.h>
+#include <assert.h>
+#include <time.h>
+#include <string.h>
+#include <signal.h>
+
+#include "rvn.h"
+#include "status_codes.h"
+#include "internal_posix.h"
+
+
+#define MSEC_TO_SEC(x) (x/1000)
+#define MSEC_TO_NANOSEC(x) ((x % 1000) * 1000000)
+
+PRIVATE char** 
+parse_cmd_line(char* line)
+{
+    if (line == NULL || *line == '\0')
+        return NULL;
+
+    char** argv = malloc(strlen(line)*2);
+    char * pch;
+    int argc = 1;    
+    
+    pch = strtok (line," \t\n");
+    while (pch != NULL)
+    {        
+        argv[argc++] = pch;
+        pch = strtok (NULL, " \t\n");
+    }  
+    argv[argc] = 0;
+
+    return argv;
+}
+
+PRIVATE ssize_t 
+ReadSize(int fd, void* buffer, size_t count)
+{
+    ssize_t rv = 0;
+    while (count > 0)
+    {
+        ssize_t result = read(fd, buffer, count);
+        while ( result < 0 && errno == EINTR)
+            result = read(fd, buffer, count);
+
+        if (result > 0)
+        {
+            rv += result;
+            buffer = (uint8_t*)buffer + result;
+            count -= (size_t)result;
+        }
+        else
+        {
+            return -1;
+        }
+    }
+    return rv;
+}
+
+PRIVATE void 
+CloseIfOpen(int fd)
+{
+    if (fd >= 0)    
+        close(fd);    
+}
+
+PRIVATE ssize_t WriteSize(int fd, const void* buffer, size_t count)
+{
+    ssize_t rv = 0;
+    while (count > 0)
+    {
+        ssize_t result = write(fd, buffer, count);
+        while ( result  < 0 && errno == EINTR)
+            result = write(fd, buffer, count);
+
+        if (result > 0)
+        {
+            rv += result;
+            buffer = (const uint8_t*)buffer + result;
+            count -= (size_t)result;
+        }
+        else
+        {
+            return -1;
+        }
+    }
+    return rv;
+}
+
+PRIVATE void 
+ExitChild(int pipeToParent, int error)
+{
+    if (pipeToParent != -1)
+        WriteSize(pipeToParent, &error, sizeof(error));
+
+    _exit(error != 0 ? error : EXIT_FAILURE);
+}
+
+EXPORT int32_t
+rvn_spawn_process(const char* filename, char* cmdline, void** pid, void** standard_in, void** standard_out, int32_t* detailed_error_code) {    
+    int stdinFds[2] = {-1, -1}, stdoutFds[2] = {-1, -1}, waitForChildToExecPipe[2] = {-1, -1};
+    pid_t processId = -1;
+    int thread_cancel_state;
+    sigset_t signal_set;
+    sigset_t old_signal_set;
+
+    char **argv = parse_cmd_line(cmdline);
+    argv[0] = (char *)filename;
```

Comment:
I don't like this, better to pass `file` to `parse_cmd_line` and have it set it there.